### PR TITLE
[Jenkins] Add nuint xml logs

### PR DESCRIPTION
**Pull request by mandel-macaque**
Change the execution of the tests to output info in xml to be able to parsed by jenkins.

### CODE REVIEW

File: tests/xharness/AppRunner.cs [hunk 1]
```diff
@@ -332,6 +420,11 @@ public int Install (Log log)
 			args.Append (" -setenv=NUNIT_AUTOEXIT=true");
 			args.Append (" -argument=-app-arg:-enablenetwork");
 			args.Append (" -setenv=NUNIT_ENABLE_NETWORK=true");
+			// detect if we are using a jenkins bot.
+			if (!string.IsNullOrEmpty (Environment.GetEnvironmentVariable ("BUILD_REVISION"))
```
Comment (rolfbjarne) [inline tests/xharness/AppRunner.cs]: There's already a Harness.InWrench method, so it makes sense to extract this to a Harness.InJenkins method as well.

**Comment by mandel-macaque** [comment]
Build

**Comment by mandel-macaque** [comment]
@rolfbjarne I was debugging, the test were not being executed as I wanted in jenkins so I wanted to see if the message was added. Will remove if it is just garbage.

**Comment by rolfbjarne** [comment]
Yeah, the `@MonkeyWrench` prefix doesn't make sense. Also it only works on wrench 😄

**Comment by mandel-macaque** [comment]
@rolfbjarne learning by touching ;) Lets see this next build, I have high hopes. If everything works, we should have much better comments from the bot.

**Comment by rolfbjarne** [comment]
The watchOS tests are still crashing.

File: tests/xharness/AppRunner.cs [hunk 2]
```diff
@@ -309,6 +311,123 @@ public int Install (Log log)
 			}
 		}
 
+		void GenerateHumanReadableLogs (string finalPath, string logHeader, XmlDocument doc){
```
Comment (rolfbjarne) [inline tests/xharness/AppRunner.cs]: style: Opening brace goes on the next line.

File: tests/xharness/AppRunner.cs [hunk 3]
```diff
@@ -309,6 +311,123 @@ public int Install (Log log)
...
+			using (var writer = new StreamWriter (finalPath)) {
+				writer.Write (logHeader);
+ 	   			var xslt = new XslCompiledTransform ();
+				xslt.Load (xsltReader);
+				xslt.Transform (xmlReader, null, writer);
+				writer.Flush ();
+			}
+		}
+
+		public bool TestsSucceeded (LogStream listener_log, bool timed_out, bool crashed) {
```
Comment (rolfbjarne) [inline tests/xharness/AppRunner.cs]: style: Opening brace goes on the next line.

File: tests/xharness/AppRunner.cs [hunk 4]
```diff
@@ -555,46 +678,7 @@ public int Install (Log log)
...
-					} else {
-						Harness.LogWrench ("@MonkeyWrench: AddSummary: {0} succeeded: {1}<br/>", mode, tests_run);
-						main_log.WriteLine ("Test run succeeded");
-						success = true;
-					}
-				} else if (timed_out) {
-					Harness.LogWrench ("@MonkeyWrench: AddSummary: <b><i>{0} timed out</i></b><br/>", mode);
-					success = false;
-				} else if (launch_failure) {
-					Harness.LogWrench ("@MonkeyWrench: AddSummary: <b><i>{0} failed to launch</i></b><br/>", mode);
```
Comment (rolfbjarne) [inline tests/xharness/AppRunner.cs]: You've lost this `launch_failure` condition (it was added after you started your branch, and probably got removed at some when you rebased).

Comment (mandel-macaque) [inline tests/xharness/AppRunner.cs]: Good catch. Fixing.

File: tests/xharness/AppRunner.cs [hunk 5]
```diff
@@ -555,50 +680,15 @@ public int Install (Log log)
...
-					crashed = true;
-					success = false;
-				}
+				success = TestsSucceeded (listener_log, timed_out, crashed);
 			} else if (timed_out) {
 				Harness.LogWrench ("@MonkeyWrench: AddSummary: <b><i>{0} never launched</i></b><br/>", mode);
 				main_log.WriteLine ("Test run never launched");
 				success = false;
+			} else if (launch_failure) {
+ 					Harness.LogWrench ("@MonkeyWrench: AddSummary: <b><i>{0} failed to launch</i></b><br/>", mode);
```
Comment (rolfbjarne) [inline tests/xharness/AppRunner.cs]: The indentation looks wrong here.

**Pull request merged**